lms/djangoapps/teams/tests/test_csv.py

```diff
@@ -330,6 +343,113 @@ def test_user_moved_to_another_team(self):
         self.assert_learner_removed_emitted(team_1.team_id, audit_learner.id)
         self.assert_learner_added_emitted(team_2.team_id, audit_learner.id)
 
+    def test_exceed_max_size(self):
+        # Given a bunch of students enrolled in a course
+        users = []
+        for i in range(5):
+            user = UserFactory.create(username='max_size_{id}'.format(id=i))
+            CourseEnrollmentFactory.create(user=user, course_id=self.course.id, mode='audit')
+            users.append(user)
+
+        # When a team is already near capaciy
+        for i in range(2):
+            user = users[i]
+            row = {'user': user, 'teamset_1': 'team_1', 'mode': 'audit'}
+            self.import_manager.add_user_to_team(row)
+
+        # ... and I try to add members in excess of capacity
+        csv_data = self._csv_reader_from_array([
+            ['user', 'mode', 'teamset_1'],
+            ['max_size_2', 'audit', 'team_1'],
+            ['max_size_3', 'audit', 'team_1']
+        ])
+
+        result = self.import_manager.set_team_memberships(csv_data)
+
+        # Then membership size is exceeded and the import fails with a "team is full" error
+        self.assertFalse(result)
+        self.assertEqual(self.import_manager.validation_errors[0], 'Team team_1 is full.')
+
+    def test_remove_from_team(self):
+        # Given a user already in a course and on a team
+        user = UserFactory.create(username='learner_1')
+        mode = 'audit'
+        CourseEnrollmentFactory.create(user=user, course_id=self.course.id, mode=mode)
+        team = CourseTeamFactory(course_id=self.course.id, name='team_1', topic_id='teamset_1')
+        team.add_user(user)
+        self.assertTrue(CourseTeamMembership.is_user_on_team(user, team))
+
+        # When I try to remove them from the team
+        csv_data = self._csv_reader_from_array([
+            ['user', 'mode', 'teamset_1'],
+            [user.username, mode, ''],
+        ])
+        result = self.import_manager.set_team_memberships(csv_data)
+
+        # Then they are removed from the team
+        self.assertFalse(CourseTeamMembership.is_user_on_team(user, team))
+
+    def test_switch_memberships(self):
+        # Given a bunch of students enrolled in a course
```

Comment:
Could you also write a test like this that fails? where you both remove and add users from a team, and fail becuase we've exceeded capacity?

Reply:
Modified `test_exceed_max_size` to include a team switch.